### ISSUE TITLE
[Device Support Request] Philips Hue Wall Switch Module

### ISSUE THREAD

**Issue by aleex1848** [zigpy/zha-device-handlers#858]
**Describe the solution you'd like**
There should be a zha_event triggered when the switch is pushed.

**Device signature - this can be acquired by removing the device from ZHA and pairing it again from the add devices screen. Be 

Log of the device joining the network:
```python
Device 0x2ab8 (00:17:88:01:0b:00:12:18) joined the network
[0x2ab8:zdo] ZDO request ZDOCmd.Device_annce: [0x2AB8, 00:17:88:01:0b:00:12:18, 128]
[0x2ab8] Requesting 'Node Descriptor'
Tries remaining: 2
[0x2ab8] Extending timeout for 0x52 request
[0x2ab8] Node Descriptor: NodeDescriptor(byte1=2, byte2=64, mac_capability_flags=128, manufacturer_code=4107, maximum_buffer_size=82, maximum_incoming_transfer_size=82, server_mask=11264, maximum_outgoing_transfer_size=82, descriptor_capability_field=0, *allocate_address=True, *complex_descriptor_available=False, *is_alternate_pan_coordinator=False, *is_coordinator=False, *is_end_device=True, *is_full_function_device=False, *is_mains_powered=False, *is_receiver_on_when_idle=False, *is_router=False, *is_security_capable=False, *is_valid=True, *logical_type=<LogicalType.EndDevice: 2>, *user_descriptor_available=False)
[0x2ab8] Discovering endpoints
Tries remaining: 3
[0x2ab8] Extending timeout for 0x53 request
[0x2ab8] Discovered endpoints: [1]
[0x2ab8:1] Discovering endpoint information
Tries remaining: 3
[0x2ab8] Extending timeout for 0x54 request
[0x2ab8:1] Discovered endpoint information: SizePrefixedSimpleDescriptor(endpoint=1, profile=260, device_type=2080, device_version=1, input_clusters=[0, 1, 3, 64512], output_clusters=[3, 4, 6, 8, 25])
[0x2ab8] Extending timeout for 0x55 request
light.osram_classic_b40_tw_lightify_e287a900_level_light_color_on_off: polling current state
[0x5a9a:3:0x0006] ZCL deserialize: <ZCLHeader frame_control=<FrameControl frame_type=GLOBAL_COMMAND manufacturer_specific=False is_reply=True disable_default_response=True> manufacturer=None tsn=86 command_id=Command.Read_Attributes_rsp>
[0x5a9a:3:0x0008] ZCL deserialize: <ZCLHeader frame_control=<FrameControl frame_type=GLOBAL_COMMAND manufacturer_specific=False is_reply=True disable_default_response=True> manufacturer=None tsn=87 command_id=Command.Read_Attributes_rsp>
[0x5A9A:3:0x0008]: received attribute: 0 update with value: 26
[0x5a9a:3:0x0300] ZCL deserialize: <ZCLHeader frame_control=<FrameControl frame_type=GLOBAL_COMMAND manufacturer_specific=False is_reply=True disable_default_response=True> manufacturer=None tsn=88 command_id=Command.Read_Attributes_rsp>
[0x2ab8:1:0x0000] ZCL deserialize: <ZCLHeader frame_control=<FrameControl frame_type=GLOBAL_COMMAND manufacturer_specific=False is_reply=True disable_default_response=True> manufacturer=None tsn=85 command_id=Command.Read_Attributes_rsp>
[0x2ab8:1] Manufacturer: Signify Netherlands B.V.
[0x2ab8:1] Model: RDM001
Checking quirks for Signify Netherlands B.V. RDM001 (00:17:88:01:0b:00:12:18)
Considering <class 'zhaquirks.xbee.xbee_io.XBeeSensor'>
Fail because endpoint list mismatch: {232, 230} {1}
Considering <class 'zhaquirks.xbee.xbee3_io.XBee3Sensor'>
Fail because endpoint list mismatch: {232, 230} {1}
Considering <class 'zhaquirks.smartthings.tag_v4.SmartThingsTagV4'>
Fail because device_type mismatch on at least one endpoint
Considering <class 'zhaquirks.smartthings.multi.SmartthingsMultiPurposeSensor'>
Fail because device_type mismatch on at least one endpoint
Considering <class 'zhaquirks.netvox.z308e3ed.Z308E3ED'>
Fail because device_type mismatch on at least one endpoint
Considering <class 'zhaquirks.gledopto.soposhgu10.SoposhGU10'>
Fail because endpoint list mismatch: {11, 13} {1}
Considering <class 'bellows.zigbee.application.EZSPCoordinator'>
Fail because device_type mismatch on at least one endpoint
device - 0x2AB8:00:17:88:01:0b:00:12:18 entering async_device_initialized - is_new_join: True
device - 0x2AB8:00:17:88:01:0b:00:12:18 has joined the ZHA zigbee network
[0x2AB8](RDM001): started configuration
[0x2AB8:ZDO](RDM001): 'async_configure' stage succeeded
[0x2ab8] Extending timeout for 0x59 request
[0x2ab8] Extending timeout for 0x5a request
[0x2ab8] Extending timeout for 0x5b request
[0x2AB8:1:0x0000]: bound 'basic' cluster: Status.SUCCESS
[0x2AB8:1:0x0000]: finished channel configuration
[0x2AB8:1:0x0001]: bound 'power' cluster: Status.SUCCESS
[0x2ab8] Extending timeout for 0x5c request
[0x2ab8] Extending timeout for 0x5d request
[0x2ab8:1:0x0001] ZCL deserialize: <ZCLHeader frame_control=<FrameControl frame_type=GLOBAL_COMMAND manufacturer_specific=False is_reply=True disable_default_response=False> manufacturer=None tsn=92 command_id=Command.Configure_Reporting_rsp>
[0x2AB8:1:0x0001]: reporting 'battery_voltage' attr on 'power' cluster: 3600/10800/1: Result: '[[ConfigureReportingResponseRecord(status=0)]]'
[0x2ab8] Extending timeout for 0x5e request
[0x2AB8:1:0xfc00]: bound 'manufacturer_specific' cluster: Status.SUCCESS
[0x2AB8:1:0xfc00]: finished channel configuration
[0x2ab8] Extending timeout for 0x5f request
[0x2ab8:1:0x0001] ZCL deserialize: <ZCLHeader frame_control=<FrameControl frame_type=GLOBAL_COMMAND manufacturer_specific=False is_reply=True disable_default_response=False> manufacturer=None tsn=94 command_id=Command.Configure_Reporting_rsp>
[0x2AB8:1:0x0001]: reporting 'battery_percentage_remaining' attr on 'power' cluster: 3600/10800/1: Result: '[[ConfigureReportingResponseRecord(status=0)]]'
[0x2AB8:1:0x0001]: finished channel configuration
[0x2ab8] Extending timeout for 0x60 request
[0x2AB8:1:0x0008]: bound 'level' cluster: Status.SUCCESS
[0x2AB8:1:0x0008]: finished channel configuration
[0x2AB8:1:0x0006]: bound 'on_off' cluster: Status.SUCCESS
[0x2AB8:1:0x0006]: finished channel configuration
[0x2AB8:1:0x0019]: bound 'ota' cluster: Status.SUCCESS
[0x2AB8:1:0x0019]: finished channel configuration
[0x2AB8:1:0x0000]: 'async_configure' stage succeeded
[0x2AB8:1:0x0001]: 'async_configure' stage succeeded
[0x2AB8:1:0xfc00]: 'async_configure' stage succeeded
[0x2AB8:1:0x0008]: 'async_configure' stage succeeded
[0x2AB8:1:0x0006]: 'async_configure' stage succeeded
[0x2AB8:1:0x0019]: 'async_configure' stage succeeded
[0x2AB8](RDM001): completed configuration
[0x2AB8](RDM001): stored in registry: ZhaDeviceEntry(name='Signify Netherlands B.V. RDM001', ieee='00:17:88:01:0b:00:12:18', last_seen=1618816060.5545013)
[0x2ab8] Extending timeout for 0x61 request
[0x2ab8:1:0x0003] ZCL deserialize: <ZCLHeader frame_control=<FrameControl frame_type=GLOBAL_COMMAND manufacturer_specific=False is_reply=True disable_default_response=False> manufacturer=None tsn=97 command_id=Command.Default_Response>
[0x2AB8:1:0x0003]: executed 'trigger_effect' command with args: '(2, 0)' kwargs: '{}' result: [64, <Status.UNSUP_CLUSTER_COMMAND: 129>]
[0x2AB8](RDM001): started initialization
[0x2AB8:ZDO](RDM001): 'async_initialize' stage succeeded
[0x2AB8:1:0x0000]: initializing channel: from_cache: False
[0x2AB8:1:0x0000]: finished channel configuration
[0x2AB8:1:0x0001]: initializing channel: from_cache: False
[0x2ab8] Extending timeout for 0x62 request
[0x2AB8:1:0xfc00]: initializing channel: from_cache: False
[0x2AB8:1:0xfc00]: finished channel configuration
[0x2AB8:1:0x0008]: initializing channel: from_cache: False
[0x2AB8:1:0x0008]: finished channel configuration
[0x2AB8:1:0x0006]: initializing channel: from_cache: False
[0x2AB8:1:0x0006]: finished channel configuration
[0x2AB8:1:0x0019]: initializing channel: from_cache: False
[0x2AB8:1:0x0019]: finished channel configuration
[0x2ab8:1:0x0001] ZCL deserialize: <ZCLHeader frame_control=<FrameControl frame_type=GLOBAL_COMMAND manufacturer_specific=False is_reply=True disable_default_response=True> manufacturer=None tsn=98 command_id=Command.Read_Attributes_rsp>
[0x2ab8] Extending timeout for 0x63 request
[0x2ab8:1:0x0001] ZCL deserialize: <ZCLHeader frame_control=<FrameControl frame_type=GLOBAL_COMMAND manufacturer_specific=False is_reply=True disable_default_response=True> manufacturer=None tsn=99 command_id=Command.Read_Attributes_rsp>
[0x2AB8:1:0x0001]: finished channel configuration
[0x2AB8:1:0x0000]: 'async_initialize' stage succeeded
[0x2AB8:1:0x0001]: 'async_initialize' stage succeeded
[0x2AB8:1:0xfc00]: 'async_initialize' stage succeeded
[0x2AB8:1:0x0008]: 'async_initialize' stage succeeded
[0x2AB8:1:0x0006]: 'async_initialize' stage succeeded
[0x2AB8:1:0x0019]: 'async_initialize' stage succeeded
[0x2AB8](RDM001): power source: Battery or Unknown
[0x2AB8](RDM001): completed initialization
```

**Comment by Hedda** [comment]
Can you check and verify if that Philips Hue Wall Switch Module is a ZGP (Zigbee Green Power) device or not?

I am guessing that it is and if so then you should know that ZGP (Zigbee Green Power) is not yet supported by zigpy/ZHA, see:

https://github.com/zigpy/zigpy/issues/341

Developers are however working on it in https://github.com/zigpy/zigpy/pull/519 and https://github.com/zigpy/zigpy/pull/656

**Comment by aleex1848** [comment]
> Can you check and verify if that Philips Hue Wall Switch Module is a ZGP (Zigbee Green Power) device or not?
> 
> I am guessing that it is and if so then you should know that ZGP is not yet supported by zigpy (or ZHA), see: [zigpy/zigpy#341](https://github.com/zigpy/zigpy/issues/341)
> 
> Developers are however working on it in [zigpy/zigpy#519](https://github.com/zigpy/zigpy/pull/519) and [zigpy/zigpy#656](https://github.com/zigpy/zigpy/pull/656)

I'm sorry, but i've no clue how to check if this device is ZGP. Never heard about that before.
On the main product page there is no info about ZGP https://www.philips-hue.com/de-de/p/hue-philips-hue-wall-switch-modul/8719514318045#specifications

**Comment by Hedda** [comment]
> On the main product page there is no info about ZGP https://www.philips-hue.com/de-de/p/hue-philips-hue-wall-switch-modul/8719514318045#specifications

Ah, that specific device (Zigbee ID "RDM001") is always connected to mains-power and that means that it would never be a ZGP.

https://zigbee.blakadder.com/Philips_RDM001.html

https://zigbeealliance.org/zigbee_products/philips-hue-wall-switch-module/

ZGP are low-energy devices that normally are battery-operated by small coin-cell battery or use energy harvesting technology.

So you can ignore my comments and links about ZGP in regards to that device.

**Comment by Adminiuga** [comment]
Do you get anything in the logs when you press buttons? Post the logs when you turn the switch on/off.

**Comment by aleex1848** [comment]
> Do you get anything in the logs when you press buttons? Post the logs when you turn the switch on/off.

I activated debug logging like this:
https://www.home-assistant.io/integrations/zha/#debug-logging

But unfortunately there is nothing logged when i push the button.

**Comment by Adminiuga** [comment]
Well, if device doesn't send anything, then the quirk unlikely to help. 
Can you try the following:
1. Unbind remote from the coordinator: devices -> your remote -> Manage zigbee clusters -> device binding -> pick your coordinator (what radio are you using) -> unbind. After you press unbind, wake up the remove by pressing any button on it
2. check what groups your coordinator is member of?
3. Bind the remote the one of the groups that coordinator is member of

post the separate logs of unbinding and binding. 
Alternatively you could put the lights you'd like to control with that remote into a new group and bind remote to that group and see if it control the lights

**Comment by jonas18z** [comment]
Yes, this would be greate if wall switch module is supported

**Comment by dieneuser** [comment]
@Adminiuga: Perhaps the following links will help. Both deconz and zigbee2mqtt support the switch:
deconz: [https://github.com/dresden-elektronik/deconz-rest-plugin/issues/4566]
zigbee2mqtt: [https://github.com/Koenkk/zigbee-herdsman-converters/issues/2393]

@Hedda: It's not a ZGP device.

**Comment by Adminiuga** [comment]
There's nothing specific in those other implementations. 
Out of the box zha should originate the events.

**Comment by dieneuser** [comment]
@Adminiuga "

> Nothing specific

is probably not it. 
I got a wall switch and can register it with ZHA, but only see a sensor with battery status but i cannot see any ZHA event when the switch is pressed. But i can see much of "unknown cluster-specific commands" in the log:
```
2021-05-14 16:38:44 DEBUG (MainThread) [zigpy.zcl] [0xb6ad:1:0x0400] ZCL deserialize: <ZCLHeader frame_control=<FrameControl frame_type=GLOBAL_COMMAND manufacturer_specific=False is_reply=True disable_default_response=True> manufacturer=None tsn=45 command_id=Command.Report_Attributes>
2021-05-14 16:38:44 DEBUG (MainThread) [zigpy.zcl] [0xb6ad:1:0x0400] ZCL request 0x000a: [[Attribute(attrid=0, value=<TypeValue type=uint16_t, value=27316>)]]
2021-05-14 16:38:44 DEBUG (MainThread) [zigpy.zcl] [0xb6ad:1:0x0400] Attribute report received: measured_value=27316
2021-05-14 16:38:52 DEBUG (MainThread) [zigpy.zcl] [0x2e33:1:0x0400] ZCL deserialize: <ZCLHeader frame_control=<FrameControl frame_type=GLOBAL_COMMAND manufacturer_specific=False is_reply=True disable_default_response=True> manufacturer=None tsn=106 command_id=Command.Report_Attributes>
2021-05-14 16:38:52 DEBUG (MainThread) [zigpy.zcl] [0x2e33:1:0x0400] ZCL request 0x000a: [[Attribute(attrid=0, value=<TypeValue type=uint16_t, value=39043>)]]
2021-05-14 16:38:52 DEBUG (MainThread) [zigpy.zcl] [0x2e33:1:0x0400] Attribute report received: measured_value=39043
2021-05-14 16:38:55 DEBUG (MainThread) [zigpy.zcl] [0x655a:1:0xfc00] ZCL deserialize: <ZCLHeader frame_control=<FrameControl frame_type=CLUSTER_COMMAND manufacturer_specific=True is_reply=True disable_default_response=True> manufacturer=4107 tsn=19 command_id=0>
2021-05-14 16:38:55 WARNING (MainThread) [zigpy.zcl] [0x655a:1:0xfc00] Unknown cluster-specific command 0
2021-05-14 16:38:55 DEBUG (MainThread) [zigpy.zcl] [0x655a:1:0xfc00] ZCL request 0x0000: b'\x01\x00\x000\x00!\x00\x00'
2021-05-14 16:38:55 DEBUG (MainThread) [zigpy.zcl] [0x655a:1:0xfc00] No handler for cluster command 0
2021-05-14 16:38:55 DEBUG (MainThread) [zigpy.zcl] [0x655a:1:0xfc00] ZCL deserialize: <ZCLHeader frame_control=<FrameControl frame_type=CLUSTER_COMMAND manufacturer_specific=True is_reply=True disable_default_response=True> manufacturer=4107 tsn=20 command_id=0>
2021-05-14 16:38:55 WARNING (MainThread) [zigpy.zcl] [0x655a:1:0xfc00] Unknown cluster-specific command 0
2021-05-14 16:38:55 DEBUG (MainThread) [zigpy.zcl] [0x655a:1:0xfc00] ZCL request 0x0000: b'\x01\x00\x000\x02!\x01\x00'
2021-05-14 16:38:55 DEBUG (MainThread) [zigpy.zcl] [0x655a:1:0xfc00] No handler for cluster command 0
2021-05-14 16:38:55 DEBUG (MainThread) [zigpy.zcl] [0x655a:1:0xfc00] ZCL deserialize: <ZCLHeader frame_control=<FrameControl frame_type=CLUSTER_COMMAND manufacturer_specific=True is_reply=True disable_default_response=True> manufacturer=4107 tsn=21 command_id=0>
2021-05-14 16:38:55 WARNING (MainThread) [zigpy.zcl] [0x655a:1:0xfc00] Unknown cluster-specific command 0
2021-05-14 16:38:55 DEBUG (MainThread) [zigpy.zcl] [0x655a:1:0xfc00] ZCL request 0x0000: b'\x01\x00\x000\x00!\x00\x00'
2021-05-14 16:38:55 DEBUG (MainThread) [zigpy.zcl] [0x655a:1:0xfc00] No handler for cluster command 0
2021-05-14 16:38:55 DEBUG (MainThread) [zigpy.zcl] [0x655a:1:0xfc00] ZCL deserialize: <ZCLHeader frame_control=<FrameControl frame_type=CLUSTER_COMMAND manufacturer_specific=True is_reply=True disable_default_response=True> manufacturer=4107 tsn=22 command_id=0>
2021-05-14 16:38:55 WARNING (MainThread) [zigpy.zcl] [0x655a:1:0xfc00] Unknown cluster-specific command 0
2021-05-14 16:38:55 DEBUG (MainThread) [zigpy.zcl] [0x655a:1:0xfc00] ZCL request 0x0000: b'\x01\x00\x000\x02!\x01\x00'
2021-05-14 16:38:55 DEBUG (MainThread) [zigpy.zcl] [0x655a:1:0xfc00] No handler for cluster command 0
2021-05-14 16:38:56 DEBUG (MainThread) [zigpy.zcl] [0x655a:1:0xfc00] ZCL deserialize: <ZCLHeader frame_control=<FrameControl frame_type=CLUSTER_COMMAND manufacturer_specific=True is_reply=True disable_default_response=True> manufacturer=4107 tsn=23 command_id=0>
2021-05-14 16:38:56 WARNING (MainThread) [zigpy.zcl] [0x655a:1:0xfc00] Unknown cluster-specific command 0
2021-05-14 16:38:56 DEBUG (MainThread) [zigpy.zcl] [0x655a:1:0xfc00] ZCL request 0x0000: b'\x01\x00\x000\x00!\x00\x00'
2021-05-14 16:38:56 DEBUG (MainThread) [zigpy.zcl] [0x655a:1:0xfc00] No handler for cluster command 0
2021-05-14 16:38:56 DEBUG (MainThread) [zigpy.zcl] [0xb4d4:1:0x0019] ZCL deserialize: <ZCLHeader frame_control=<FrameControl frame_type=CLUSTER_COMMAND manufacturer_specific=False is_reply=False disable_default_response=False> manufacturer=None tsn=162 command_id=1>
2021-05-14 16:38:56 DEBUG (MainThread) [zigpy.zcl] [0xb4d4:1:0x0019] ZCL req
```

**Comment by aleex1848** [comment]
I just updated from core-2021.4.3 to core-2021.6.2 but unfortunately nothing changed. There are still no zha_events when pushing the hue wall switch.

**Comment by jonas18z** [comment]
@dieneuser I have exactly the same problem as you I can ad but only find the battery level. I have a hue dimmer which I can find all different buttons. Anyone figure ut how to use this? 

I have also updated the firmware on my conbeeII

**Comment by dieneuser** [comment]
@jonas18z ZHA does not (yet) support the switch. As a workaround, I use the Deconz integration in parallel - also for existing Hue Tap switch modules. There, the wall switch is already supported in the beta phoscon web app.

**Comment by jonas18z** [comment]
Ok, let’s hope that it will be supported. It’s only ZHA that misses the support now. I don’t think you can run conbee II at the same time in ZHA and deconz?

**Comment by htodep** [comment]
How/where can we request ZHA support? Really need this device ;)

**Comment by dieneuser** [comment]
> I don’t think you can run conbee II at the same time in ZHA and deconz?
That's right. I use a second Conbee II with deconz integration.

**Comment by jonas18z** [comment]
Is there some guide how to add new devices. I will try to add it I was thinking to just modify the hue dimmer switch file to make this wall switch to work. But I don’t find anything here

**Comment by MattWestb** [comment]
https://github.com/zigpy/zha-device-handlers#how-to-contribute

**Comment by jonas18z** [comment]
Thanks, Installed 2021.9 and did a new search for device and the device is found and added via ZHA but now next problem is that it only founds two triggers "Device Offline" and "power battery level changes" For example my dimmer RWL021 founds all trigger on,off, dim up, dim down etc with RDM001. So maybe there is some more editing needed in the Python file but I can't find what?

**Comment by MarvinSchenkel** [comment]
> Thanks, Installed 2021.9 and did a new search for device and the device is found and added via ZHA but now next problem is that it only founds two triggers "Device Offline" and "power battery level changes" For example my dimmer RWL021 founds all trigger on,off, dim up, dim down etc with RDM001. So maybe there is some more editing needed in the Python file but I can't find what?
> 
> EDIT: I don't se any zha_event when pushing a button.

I have the exact same issue. Has anyone found the reason for this?

**Comment by jonas18z** [comment]
I have the same issue if some know what could be wrong or what I could try to make it work.

**Comment by florianw12** [comment]
Hello I think I have the same issue but I am not sure. I am using a raspberry pi 4 with homeassistant in a docker container and the conbee 2 usb on the raspberry with deconz also in a container. I have managed to connect the hue wall switch and I get events in home assistant but I am not able to use them for anything. Wall switch also shows just as battary in devices
![image](https://user-images.githubusercontent.com/91988937/141531089-212db4c9-0531-415c-963b-61371a3238ad.png)

**Comment by MattWestb** [comment]
In ZHA is normal switches (or if battery powered remotes) only getting one battery.
All things one switch is doing event and you is getting them by development, events, listening to events and putting in `zha_event` and clicking start listening to events and you is getting all in the log window.

One remote dont have any state like one light that can being on or off or dimmed and only sending events.

Also if Philips have making it good it shall having one switch configure cluster for changing the function of the wall switch but i cant saying if its having it then no one have posting one device signature.

**Comment by MattWestb** [comment]
If some like implanting the switch configuration i think i have found the attribute that need being added in one quirk for getting it working.
https://github.com/dresden-elektronik/deconz-rest-plugin/blob/3338039cb8a498f90f5000f0b118ded7ef963a84/general.xml#L211-L229

**Comment by wigster** [comment]
> and you is getting them by development, events, listening to events and putting in `zha_event` and clicking start listening to events and you is getting all in the log window.

So does this mean that XHA supports the module or not yet? I was about to purchase a few but am hesitating.

**Comment by water-escape** [comment]
Same here. Using ZHA. Did anyone get this to work via ZHA so it is useable in automations.

**Comment by JoachimCanu** [comment]
Hello,

I see that this subject is not moving forward. the same as the Aqara MCCGQ14LM.
The only solution is to learn to contribute by integrating new products.
I have read the README.md : zigpy/zha-device-handlers

Is there a tutorial ? (if possible in FR, but English will do just as well)

**Comment by tube0013** [comment]
Worked with @dmulcahey and got a rough quirk working for this device. device automation triggers are only for the 1st input.

still some thought to be done on how best to implement this in the code base, but this is usable as a custom zha quirk just drop the py file in the custom quirk folder as defined in your configuration.yaml under zha. <https://www.home-assistant.io/integrations/zha/#custom_quirks_path>

to enable the second input - go to the devices configuration page > Manage Clusters.

pick the Philips Basic cluster - pick the mode attribute, put 2 in the value and set attirbute (toggle switch to wake the device so it gets the setting.

![image](https://user-images.githubusercontent.com/11084412/145148459-b3dd0a9b-0ead-47c4-b45b-f0f278ff5b17.png)




[rdm001.py.zip](https://github.com/zigpy/zha-device-handlers/files/7673371/rdm001.py.zip)

**Comment by dieneuser** [comment]
@tube0013 

Thank you for the quirk file.
But I can't manage to configure the switch to double rocker/Dual Push Button. No mode value (0,1,2,3) is taken over in the cluster manage window. When reading out the value is always enum8.undefined_0x00 or None. The switch remains in the default configuration as Single Rocker - but I really need the Dual Push Button mode (value=3).

here all possible value for the mode:
mode value="0" type="Single Rocker"
mode value="1" type="Single Push Button"
mode value="2" type="Dual Rocker"
mode value="3" type="Dual Push Button"

**Comment by tube0013** [comment]
Was the switch awake when the attribute to change the mode is written? May want to enable debug logging and watch to be sure the attribute is received. I confirmed changing to mode 2 worked but did not try any others.

**Comment by dieneuser** [comment]
Oh, thanks for pointing out the debug level. 
```
2021-12-11 16:39:59 DEBUG (MainThread) [homeassistant.components.zha.core.device] [0x432F](RDM001): set: None for attr: 52 to cluster: 0 for ept: 1 - res: [2, <Status.MALFORMED_COMMAND: 128>]
2021-12-11 16:39:59 DEBUG (MainThread) [homeassistant.components.zha.api] Set attribute for: cluster_id: [0] cluster_type: [in] endpoint_id: [1] attribute: [52] value: [None] manufacturer: [2] response: [[2, <Status.MALFORMED_COMMAND: 128>]]
```
I wrote the value in the wrong line in the dialog (manufacturer code), so the mode value is None. 

Now it worked. I use value 3 for Dual Push Button mode and it works perfect. 👍

**Comment by htodep** [comment]
@tube0013 Thank you so much! it worked like a charm.
Is this a permanent fix or does it need some tweaking(works fine though)?

**Comment by aleex1848** [comment]
> Worked with @dmulcahey and got a rough quirk working for this device. device automation triggers are only for the 1st input.
> 
> still some thought to be done on how best to implement this in the code base, but this is usable as a custom zha quirk just drop the py file in the custom quirk folder as defined in your configuration.yaml under zha. https://www.home-assistant.io/integrations/zha/#custom_quirks_path
> 
> to enable the second input - go to the devices configuration page > Manage Clusters.
> 
> pick the Philips Basic cluster - pick the mode attribute, put 2 in the value and set attirbute (toggle switch to wake the device so it gets the setting.
> 
> ![image](https://user-images.githubusercontent.com/11084412/145148459-b3dd0a9b-0ead-47c4-b45b-f0f278ff5b17.png)
> 
> [rdm001.py.zip](https://github.com/zigpy/zha-device-handlers/files/7673371/rdm001.py.zip)

After many months  i'm finaly able to use the switch module. Thank you so much!!

**Comment by WernerRH21** [comment]
@tube0013 
I am excited to find this solution, but unfortunately don't know exactly what to do. Could you please explain your sentence "just drop the py file in the custom quirk folder as defined in your configuration.yaml under zha" a littlebit more? I don't have a zha section in my configuration.yaml
Thanks in advance

**Comment by tube0013** [comment]
you add the zha: section like this:

```
zha:
  custom_quirks_path: zha_custom_quirks
```

then in your config folder create a folder named zha_custom_quirks and within that drop the .py file that's in the zip I posted:

![image](https://user-images.githubusercontent.com/11084412/148699197-688b6109-4b72-49ed-aeed-8657bec2b7d4.png)

**Comment by WernerRH21** [comment]
Thx for your quick reply.
Have the zip file smoothly overlooked :) Now the section entry is set and the file is in place and I see the PhilipsBasicCluster. Sorry for my maybe silly questions. But what do I have to do now? I come from ioBroker and are quite new to HA. Thanks for your patience.

**Comment by WernerRH21** [comment]
This is all I need, or shall they also have an entity?
![image](https://user-images.githubusercontent.com/67897090/148700522-6474ec04-9577-4d61-88cc-43ee7f9347d6.png)

**Comment by WernerRH21** [comment]
I created an automation to toggle a hue lamp when "Turn on" button pressed, but no success. Same automation with an Aqara Switch runs as expected.

**Comment by htodep** [comment]
@WernerRH21 what kind of switch are you using? Don't forget to change the cluster settings as mentioned above if you have a double switch or something else. Check the logs when you press the switch and see what triggers com up. For automation use remote_button_short_press as type and turn_on as subtype.

**Comment by WernerRH21** [comment]
A single push button switch and set the cluster settings like below and push the switch when I set zigbee attribute. "Set Zigbee attributes" blinks green.
![image](https://user-images.githubusercontent.com/67897090/148703169-b9ffc931-0091-4219-bdba-98a20da05406.png)
Where can I check the log for this device? In the logbook is no entry when I select Hue Wall Switch entity.

**Comment by htodep** [comment]
https://www.home-assistant.io/integrations/zha/#debug-logging

**Comment by WernerRH21** [comment]
Hi @htodep 
Cool, thank you. I added the wall switch again and set the cluster setting. Should be fine now?
![image](https://user-images.githubusercontent.com/67897090/148744521-608368e8-32f9-4ad3-99d4-c27019c3513d.png)
Where do I find for automation remote_button_short_press as type and turn_on as subtype. 
![image](https://user-images.githubusercontent.com/67897090/148744771-629dcfff-f0e1-4655-b3fd-d78171e265c3.png)

**Comment by WernerRH21** [comment]
Amazing. I got it. Thank you so much guys. How much you can enjoy a simple smart light switch.

**Comment by htodep** [comment]
Nice! I know the feeling 😎 How did you make it work at the end?

**Comment by WernerRH21** [comment]
The debug-logging was important to see that the switch works. Then just "Turn on button" as trigger and bäähm, the light toggles. 
![image](https://user-images.githubusercontent.com/67897090/148747426-bc8403f9-24df-4eda-b953-8183e6f028db.png)

**Comment by RaveGun** [comment]
Thank you!

I also followed the examples and hints and made it work on my desk.
Now I just have to drill and mount everything.

PS. Is there a possibility to connect the Switch directly to the Bulb so that an automation is not needed?

Thanks!

**Comment by Matthiasfranck** [comment]
Thnx for this fix.

Can someone tell me with which version(s) of home assistant this is compatible? I am currently using `core-2021.11.5`.

**Comment by ark-** [comment]
I have got the custom quirks working with my wall module. What's preventing this being merged into the repo and released to the next release of HA?

**Comment by tube0013** [comment]
I'll try and get this into a proper PR so it can be put into a release.

**Comment by tube0013** [comment]
#1434

**Comment by m4dmin** [comment]
I'm trying to configure my Hue wall Switch Module as Dual Rocker, but I can't get it to work.

I've followed all the steps mentiones in this thread:

1. Added the Custom Quirk file

> you add the zha: section like this:
> 
> ```
> zha:
>   custom_quirks_path: zha_custom_quirks
> ```
> 
> then in your config folder create a folder named zha_custom_quirks and within that drop the .py file that's in the zip I posted:
> 
> ![image](https://user-images.githubusercontent.com/11084412/148699197-688b6109-4b72-49ed-aeed-8657bec2b7d4.png)


2. Enabled the Second Input Mode via Manage Cluster: 

> > Worked with @dmulcahey and got a rough quirk working for this device. device automation triggers are only for the 1st input.
> > still some thought to be done on how best to implement this in the code base, but this is usable as a custom zha quirk just drop the py file in the custom quirk folder as defined in your configuration.yaml under zha. https://www.home-assistant.io/integrations/zha/#custom_quirks_path
> > to enable the second input - go to the devices configuration page > Manage Clusters.
> > pick the Philips Basic cluster - pick the mode attribute, put 2 in the value and set attirbute (toggle switch to wake the device so it gets the setting.
> > ![image](https://user-images.githubusercontent.com/11084412/145148459-b3dd0a9b-0ead-47c4-b45b-f0f278ff5b17.png)
> > [rdm001.py.zip](https://github.com/zigpy/zha-device-handlers/files/7673371/rdm001.py.zip)
> 
> After many months i'm finaly able to use the switch module. Thank you so much!!

3. Verfied the correct setting via Debug Log:

`2022-03-24 13:40:56 DEBUG (MainThread) [homeassistant.components.zha.core.device] [0x4A45](RDM001): set: 2 for attr: 52 to cluster: 0 for ept: 1 - res: [[WriteAttributesStatusRecord(status=<Status.SUCCESS: 0>)]]`
`2022-03-24 13:40:56 DEBUG (MainThread) [homeassistant.components.zha.api] Set attribute for: cluster_id: [0] cluster_type: [in] endpoint_id: [1] attribute: [52] value: [2] manufacturer: [None] response: [[[WriteAttributesStatusRecord(status=<Status.SUCCESS: 0>)]]]`

4. Created two automations, one for the first and one for the second switch.

But in the end I can only use the first switch.

- Am I missing something?
- How can I adress the second switch within an automation?

**Comment by evpaassen** [comment]
> How can I adress the second switch within an automation?

@m4dmin I'm using an *Event* trigger to listen to a `zha_event` which I captured using the developer tools:

![image](https://user-images.githubusercontent.com/3531885/159925297-2e870999-3aef-4cc8-800e-d5d3d7c1aa65.png)

When you go into Developer tools > Events you can enter the `zha_event` type and listen to those events. If you flip your switch, this will show you the `device_ieee`.

**Comment by m4dmin** [comment]
Thanks @evpaassen. That was the missing piece :)

**Comment by RaveGun** [comment]
Hi, After the update to `Home Assistant 2022.4.0` the ZHA is not starting because of an error from the `rdm001.py`.
This is what is in the log recorded:

```
Error setting up entry ConBee II, s/n: DE2225070 - dresden elektronik ingenieurtechnik GmbH for zha
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 335, in async_setup
    result = await component.async_setup_entry(hass, self)
  File "/usr/src/homeassistant/homeassistant/components/zha/__init__.py", line 99, in async_setup_entry
    setup_quirks(config)
  File "/usr/local/lib/python3.9/site-packages/zhaquirks/__init__.py", line 409, in setup
    importer.find_module(modname).load_module(modname)
  File "<frozen importlib._bootstrap_external>", line 529, in _check_name_wrapper
  File "<frozen importlib._bootstrap_external>", line 1029, in load_module
  File "<frozen importlib._bootstrap_external>", line 854, in load_module
  File "<frozen importlib._bootstrap>", line 274, in _load_module_shim
  File "<frozen importlib._bootstrap>", line 711, in _load
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/config/zha_custom_quirks/rdm001.py", line 51, in <module>
    class PhilipsBasicCluster(CustomCluster, Basic):
  File "/usr/local/lib/python3.9/site-packages/zigpy/zcl/__init__.py", line 91, in __init_subclass__
    raise TypeError(
TypeError: `manufacturer_attributes` is deprecated. Copy the parent class's `attributes` dictionary and update it with your manufacturer-specific `attributes`. Make sure to specify that it is manufacturer-specific through the  appropriate constructor or tuple!
```

I do not understand the issue and how to modify the python file to make it work.

Any hints?

Thank you.

**Comment by BrendanxP** [comment]
> Hi, After the update to `Home Assistant 2022.4.0` the ZHA is not starting because of an error from the `rdm001.py`. This is what is in the log recorded:
> 
> ```
> Error setting up entry ConBee II, s/n: DE2225070 - dresden elektronik ingenieurtechnik GmbH for zha
> Traceback (most recent call last):
>   File "/usr/src/homeassistant/homeassistant/config_entries.py", line 335, in async_setup
>     result = await component.async_setup_entry(hass, self)
>   File "/usr/src/homeassistant/homeassistant/components/zha/__init__.py", line 99, in async_setup_entry
>     setup_quirks(config)
>   File "/usr/local/lib/python3.9/site-packages/zhaquirks/__init__.py", line 409, in setup
>     importer.find_module(modname).load_module(modname)
>   File "<frozen importlib._bootstrap_external>", line 529, in _check_name_wrapper
>   File "<frozen importlib._bootstrap_external>", line 1029, in load_module
>   File "<frozen importlib._bootstrap_external>", line 854, in load_module
>   File "<frozen importlib._bootstrap>", line 274, in _load_module_shim
>   File "<frozen importlib._bootstrap>", line 711, in _load
>   File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
>   File "<frozen importlib._bootstrap_external>", line 850, in exec_module
>   File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
>   File "/config/zha_custom_quirks/rdm001.py", line 51, in <module>
>     class PhilipsBasicCluster(CustomCluster, Basic):
>   File "/usr/local/lib/python3.9/site-packages/zigpy/zcl/__init__.py", line 91, in __init_subclass__
>     raise TypeError(
> TypeError: `manufacturer_attributes` is deprecated. Copy the parent class's `attributes` dictionary and update it with your manufacturer-specific `attributes`. Make sure to specify that it is manufacturer-specific through the  appropriate constructor or tuple!
> ```
> 
> I do not understand the issue and how to modify the python file to make it work.
> 
> Any hints?
> 
> Thank you.

Exactly the same issue here. I deleted the ZHA quirks from my configuration.yaml file and everything works now!

**Comment by tube0013** [comment]
the quirk was added to the most recent quirks release and is in the new HA release. if you installed it to custom_quirks, deleted the rdm001.py from the custom_quirks folder in your config.

**Comment by RaveGun** [comment]
Oh, that was an easy fix.
Delete I can do :)
Thank you a lot.

**Comment by maddingamer** [comment]
> > How can I adress the second switch within an automation?
> When you go into Developer tools > Events you can enter the `zha_event` type and listen to those events. If you flip your switch, this will show you the `device_ieee`.

When I flip switch 2 it does not trigger an zha_event. How do you make switch 2 work? 🤔 (Using zha on HA 2022.4)

**Comment by deridderlab** [comment]
Thanks for sharing! Managed to get both the left_press and right_press enables through setting the Cluster Attributes. 

However.... 

Anyone has the issue that an actual button press issues both a "left_press" and "left_press_release" event? And in fact, the actual button release dues the same (also a "left_press" and "left_press_release")? Not a huge issue, but unexpected behavior...

**Comment by niwi-hh** [comment]
I wonder if using the walls witch with ZHA can cause the battery to run low much faster than normally. Two of my four wall switches are out of power after using them about a month with ZHA. A third more close the my zigbee bridge is still fine as well as the fourth switch still controlled by a hue bridge. The batteries should last about 5 years. I bought the switches some months ago.

**Comment by Hedda** [comment]
Please do not stale, (I do not have these myself but Philips Hue is still one of the most popular
brands for Zigbee products so should ideally all work in ZHA).

**Comment by TheJulianJES** [comment]
Is the only issue here that the device triggers are missing for the right button if activated?
(And that we should probably have a config entity in HA for changing the mode?)

**Comment by TheJulianJES** [comment]
The right trigger buttons should be available in the UI now (2023.2.0 and later).
Due to how only the left button was added first, it's named a bit weirdly in the UI (would have been a breaking change otherwise).

Are there any remaining issues with this?

**Comment by tkreeftmeijer** [comment]
I found the following zha_event seems to have no corresponding device event in HA. Did I something wrong or miss I something? I have set mode to 2. It would be really nice if we can set it via the UI.

> event_type: zha_event
> data:
>   device_ieee: 00:17:88:01:0c:c2:80:80
>   unique_id: 00:17:88:01:0c:c2:80:80:1:0x0006
>   device_id: ee58328b287b1d7cc64ad14c5e2fe090
>   endpoint_id: 1
>   cluster_id: 6
>   command: toggle
>   args: []
>   params: {}
> origin: LOCAL
> time_fired: "2023-02-04T16:58:40.592149+00:00"
> context:
>   id: 01GRENV82GM0410GKKN2DS5604
>   parent_id: null
>   user_id: null

**Comment by TheJulianJES** [comment]
That's an event for when the module is directly bound to a bulb. When in mode 2, it should also produce the following events:

https://github.com/zigpy/zha-device-handlers/blob/84d02be7abde55a6cee80fa155f0cbbc20347c40/zhaquirks/philips/rdm001.py#L179-L196

Do you also get these events?

**Comment by tkreeftmeijer** [comment]
> That's an event for when the module is directly bound to a bulb. When in mode 2, it should also produce the following events:
> 
> https://github.com/zigpy/zha-device-handlers/blob/84d02be7abde55a6cee80fa155f0cbbc20347c40/zhaquirks/philips/rdm001.py#L179-L196
> 
> Do you also get these events?

A complete reset (of the module and remove it from ZHA) did the job. It works now. It was not directly bound to a light, I think that was still the problem. Thanks.

**Comment by markrossington** [comment]
My Wall Switch Module on HA OS with HA2023.6 using a Skyconnect still only reports left presses and not right presses

Is this to be expected?

Got it to work by writing to the mode register manually and setting it to 3

<img width="522" alt="image" src="https://github.com/zigpy/zha-device-handlers/assets/124208527/18d726e8-823d-4acc-bbf8-41ca1ae0b77c">

**Comment by andyjenkinson** [comment]
> My Wall Switch Module on HA OS with HA2023.6 using a Skyconnect still only reports left presses and not right presses
> 
> Is this to be expected?
> 
> Got it to work by writing to the mode register manually and setting it to 3

Yes if you don't configure it otherwise, the  module is factory defaulted to mode 0 - a single rocker switch. If you have installed it connected to a double momentary switch then you have to simulate what the Hue app does, i.e. configuring the mode attribute. The switch hardware doesn't automatically know how you have it physically wired. Not sure if ZHA could implement a custom device configuration workflow in the UI like the Hue app does but for the moment AFAIK there is not one.

**Comment by mirceadb7** [comment]
I have a few Hue Wall Switch RDM004 and HA Yellow, in zigbee zha integration I don’t have the cluster: Philips Basic Cluster.

**Comment by KWOAD** [comment]
> I have a few Hue Wall Switch RDM004 and HA Yellow, in zigbee zha integration I don’t have the cluster: Philips Basic Cluster.

Correct, I also can't find that cluster in the new version of the Philips Hue Wall Switch, the RDM004. I'm waiting for ZHA supports it. Really annoying because otherwise it works fine. I looked in all the clusters, but can't find 'mode' or '0x0034'.

**Comment by Dryghor** [comment]
> > I have a few Hue Wall Switch RDM004 and HA Yellow, in zigbee zha integration I don’t have the cluster: Philips Basic Cluster.
> 
> Correct, I also can't find that cluster in the new version of the Philips Hue Wall Switch, the RDM004. I'm waiting for ZHA supports it. Really annoying because otherwise it works fine. I looked in all the clusters, but can't find 'mode' or '0x0034'.

@KWOAD and @mirceadb7 did you try to delete the py file from the custom_quirks folder ?
I had this same issue (and my automation was barely working : bulb was on then instantly off when triggered), delete it then I was able to find 'mode' attribute under 'PhilipsBasicCluster' again.

I am on HA Green with Connect ZBT-1 (SkyConnect) and RDM001

**Comment by cbornemann** [comment]
Since a few updates, the custom quirk for RDM001 is not applied to my RDM001 anymore.

I have to set up a old custom quirk so that it working:

````"""Signify RDM001 device."""
import logging
from typing import Any, List, Optional, Union

from zigpy.profiles import zha
from zigpy.quirks import CustomCluster, CustomDevice
import zigpy.types as t
from zigpy.zcl import foundation
from zigpy.zcl.clusters.general import (
    Basic,
    Groups,
    Identify,
    LevelControl,
    OnOff,
    Ota,
    PowerConfiguration,
)

from zhaquirks.const import (
    ARGS,
    BUTTON,
    COMMAND,
    COMMAND_ID,
    DEVICE_TYPE,
    DOUBLE_PRESS,
    ENDPOINTS,
    INPUT_CLUSTERS,
    LONG_PRESS,
    LONG_RELEASE,
    MODELS_INFO,
    OUTPUT_CLUSTERS,
    PRESS_TYPE,
    PROFILE_ID,
    QUADRUPLE_PRESS,
    QUINTUPLE_PRESS,
    RIGHT,
    SHORT_PRESS,
    SHORT_RELEASE,
    TRIPLE_PRESS,
    TURN_ON,
    ZHA_SEND_EVENT,
)
from zhaquirks.philips import PHILIPS, SIGNIFY

DEVICE_SPECIFIC_UNKNOWN = 64512
_LOGGER = logging.getLogger(__name__)


class PhilipsBasicCluster(CustomCluster, Basic):
    """Philips Basic cluster."""

    attributes = Basic.attributes.copy()
    attributes.update(
        {
            0x0031: ("philips", t.bitmap16, True),
            0x0034: ("mode", t.enum8, True),    
        }
    )

    attr_config = {0x0031: 0x000B, 0x0034: 0x02}

    async def bind(self):
        """Bind cluster."""
        result = await super().bind()
        await self.write_attributes(self.attr_config, manufacturer=0x100B)
        return result


class PhilipsRemoteCluster(CustomCluster):
    """Philips remote cluster."""

    cluster_id = 64512
    name = "PhilipsRemoteCluster"
    ep_attribute = "philips_remote_cluster"
    client_commands = {
        0x00: foundation.ZCLCommandDef(
            "notification",
            {
                "param1": t.uint8_t,
                "param2": t.uint24_t,
                "param3": t.uint8_t,
                "param4": t.uint8_t,
                "param5": t.uint8_t,
                "param6": t.uint8_t,
            },
            is_manufacturer_specific=True,
            direction=foundation.Direction.Server_to_Client,
        )
    }
    BUTTONS = {
        1: "left",
        2: "right",
    }
    PRESS_TYPES = {0: "press", 1: "hold", 2: "press_release", 3: "hold_release"}

    def handle_cluster_request(
        self,
        hdr: foundation.ZCLHeader,
        args: List[Any],
        *,
        dst_addressing: Optional[
            Union[t.Addressing.Group, t.Addressing.IEEE, t.Addressing.NWK]
        ] = None,
    ):
        """Handle the cluster command."""
        _LOGGER.debug(
            "PhilipsRemoteCluster - handle_cluster_request tsn: [%s] command id: %s - args: [%s]",
            hdr.tsn,
            hdr.command_id,
            args,
        )

        button = self.BUTTONS.get(args[0], args[0])
        press_type = self.PRESS_TYPES.get(args[2], args[2])

        event_args = {
            BUTTON: button,
            PRESS_TYPE: press_type,
            COMMAND_ID: hdr.command_id,
            ARGS: args,
        }

        action = f"{button}_{press_type}"
        self.listener_event(ZHA_SEND_EVENT, action, event_args)


class PhilipsRDM001(CustomDevice):
    """Philips RDM001 device."""

    signature = {
        #  <SimpleDescriptor endpoint=1 profile=260 device_type=2080
        #  device_version=1
        #  input_clusters=[0, 1, 3, 64512]
        #  output_clusters=[3, 4, 6, 8, 25]>
        MODELS_INFO: [(PHILIPS, "RDM001"), (SIGNIFY, "RDM001")],
        ENDPOINTS: {
            1: {
                PROFILE_ID: zha.PROFILE_ID,
                DEVICE_TYPE: zha.DeviceType.NON_COLOR_SCENE_CONTROLLER,
                INPUT_CLUSTERS: [
                    Basic.cluster_id,
                    PowerConfiguration.cluster_id,
                    Identify.cluster_id,
                    DEVICE_SPECIFIC_UNKNOWN,
                ],
                OUTPUT_CLUSTERS: [
                    Basic.cluster_id,
                    Identify.cluster_id,
                    Groups.cluster_id,
                    OnOff.cluster_id,
                    LevelControl.cluster_id,
                    Ota.cluster_id,
                ],
            }
        },
    }

    replacement = {
        ENDPOINTS: {
            1: {
                PROFILE_ID: zha.PROFILE_ID,
                DEVICE_TYPE: zha.DeviceType.NON_COLOR_CONTROLLER,
                INPUT_CLUSTERS: [
                    PhilipsBasicCluster,
                    PowerConfiguration.cluster_id,
                    Identify.cluster_id,
                    PhilipsRemoteCluster,
                ],
                OUTPUT_CLUSTERS: [
                    Ota.cluster_id,
                    Identify.cluster_id,
                    Groups.cluster_id,
                    OnOff.cluster_id,
                    LevelControl.cluster_id,
                ],
            }
        }
    }

    device_automation_triggers = {
        (SHORT_PRESS, TURN_ON): {COMMAND: "left_press"},
        (LONG_PRESS, TURN_ON): {COMMAND: "left_hold"},
        (DOUBLE_PRESS, TURN_ON): {COMMAND: "left_double_press"},
        (TRIPLE_PRESS, TURN_ON): {COMMAND: "left_triple_press"},
        (QUADRUPLE_PRESS, TURN_ON): {COMMAND: "left_quadruple_press"},
        (QUINTUPLE_PRESS, TURN_ON): {COMMAND: "left_quintuple_press"},
        (SHORT_RELEASE, TURN_ON): {COMMAND: "left_short_release"},
        (LONG_RELEASE, TURN_ON): {COMMAND: "left_long_release"},
        (SHORT_PRESS, RIGHT): {COMMAND: "right_press"},
        (LONG_PRESS, RIGHT): {COMMAND: "right_hold"},
        (DOUBLE_PRESS, RIGHT): {COMMAND: "right_double_press"},
        (TRIPLE_PRESS, RIGHT): {COMMAND: "right_triple_press"},
        (QUADRUPLE_PRESS, RIGHT): {COMMAND: "right_quadruple_press"},
        (QUINTUPLE_PRESS, RIGHT): {COMMAND: "right_quintuple_press"},
        (SHORT_RELEASE, RIGHT): {COMMAND: "right_short_release"},
        (LONG_RELEASE, RIGHT): {COMMAND: "right_long_release"},
    }````

**Comment by Hedda** [comment]
FYI, looks like [RyanMorash](https://github.com/RyanMorash) recently started on a related pull request here that could/should fix this(?) -> https://github.com/zigpy/zha-device-handlers/pull/4228

- https://github.com/zigpy/zha-device-handlers/pull/4228

**Comment by RyanMorash** [comment]
My pull request is what I'm currently running on my ZHA. I'm able to see battery status and use automations now. 
It seems as though a recent update changed the clusters on the RDM001 to match what the quirk was expecting in the RDM004.
